pkg/autodiscovery/listeners/snmp.go

```diff
@@ -146,7 +146,7 @@ var worker = func(l *SNMPListener, jobs <-chan snmpJob) {
 }
 
 func (l *SNMPListener) checkDevice(job snmpJob) {
-	params := *job.subnet.defaultParams
+	params := job.subnet.defaultParams
```

Comment:
As mentioned in the PR description, I had to remove a pointer dereference here to work around this `go vet` error 
```
assignment copies lock value to params: github.com/gosnmp/gosnmp.GoSNMP contains sync.Mutex
```
since gosnmp/gosnmp@cad7b2aabd35f69f7cf53b96d392a90d3035385e (present since `v1.27.0`) added a mutex to the `GoSNMP` struct.

Reply:
I think this was creating a copy of `job.subnet.defaultParams` and removing the dereference might lead to race condition.

Looks like the mutex has been removed recently in `1.31.0`  https://github.com/gosnmp/gosnmp/commit/567013a05e82e5c64bc2f67281a6bf4062089a28

Could you update to `github.com/gosnmp/gosnmp` from `1.29.0` to `1.31.0` ? and check if it works fine. I'm making some tests to check if the update is not breaking something.

Reply:
Done, bumped to v1.31.0 and reverted the change on this line